devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
@fablup Sounds good to me. This should indeed allow us to have a much more detailed "debug" mode, which sounds great.